Kickstarter-iOS/ViewModels/AppDelegateViewModel.swift

```diff
@@ -21,6 +22,12 @@ public func == (lhs: HockeyConfigData, rhs: HockeyConfigData) -> Bool {
     && lhs.userName == rhs.userName
 }
 
+public protocol NotificationAuthorizationStatusType {
+  var isAuthorized: Bool { get }
+  var isDenied: Bool { get }
+  var isNotDetermined: Bool { get }
+}
```

Comment:
It all boils down to the property 
```
fileprivate let notificationAuthorizationStatusProperty =
    MutableProperty<NotificationAuthorizationStatusType?>(nil)
  public func notificationAuthorizationStatusReceived(
    _ notificationSetting: NotificationAuthorizationStatusType
  ) {
    return self.notificationAuthorizationStatusProperty.value = notificationSetting
  }
```
of
 `AppDelegateViewModel` being specified only for iOS 10. I would prefer having it as single  enum too but couldn't come up with the way of getting away with direct usage of UNAuthorizationStatus enum